Add support for specifying non-local directories when doing system-wide file scans

matejak opened the pull request:
Initial support added to `file_permissions_unauthorized_world_writable` via the `var_blacklisted_paths` XCCDF value.
The rule scans the whole (local) filesystem, it already filters out some directory trees (`/proc`, `/sys`, `/selinux/...`), but there may be non-local filesystems that the scanner may not be able to correctly identify, and it may not be solvable on the scanner side. This PR adds a possibility for an opt-in override. See also: https://bugzilla.redhat.com/show_bug.cgi?id=1840579

The PR doesn't work out of the box - please help: for some reason, the build system doesn't add the check-export element to the rule XCCDF, without which the external variable used in the OVAL doesn't work. If I add the `check-export` manually into the datastream, it starts to work.

I think that this mechanism could be ported to macros and used in more rules to enable users to e.g. exclude `/proc` etc. What I don't know is whether this filtering could prevent scanning (which would be really nice), or just collecting, i.e. the scanner would scan, but throw away findings and not eat more RAM, or neither, when the scanner scans, collects and waits for the end to throw everything away.

mildas commented:
**Changes identified:**
Rule file_permissions_unauthorized_world_writable:
 Text added outsite tags in OVAL check.
 Node deleted from OVAL check.
 Deleted attribute from OVAL check.
 Node moved within OVAL check.
 Text changed in OVAL check.
 Attribute value changed in OVAL check.
 New node inserted to OVAL check.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_world_writable

jan-cerny commented on linux_os/guide/system/var_blacklisted_paths.var:
```diff
@@ -0,0 +1,15 @@
+documentation_complete: true
+
+title: 'Blacklisted paths'
```
I assume you want multiple rules sharing this variable. Why do you think so? Shouldn't each rule have a different variable?

matejak replied:
That's up to a discussion. My idea was to provide a way how to reliably disable a folder one is 100% sure that it shouldn't be ever scanned (dynamic FS, network FS).

jan-cerny replied:
Sounds great!

Let me add some arguments against it. I think is that people won't know which all rules are affected by this variable, so they might think it's only one or all of them. Also, unless you change all the rules you won't make sure that the scanner won't access the file system.

But I think that your proposal is better in terms of user experience, as people will tailor only a single variable. Also, it will be easier approach for the content developers.

@yuumasato @vojtapolasek Any other ideas?

yuumasato replied:
I would prefer to avoid having one general variable to skip paths, different rules may want skip a path for different reasons.

For this case, I see that intention is to list all the remote paths.
The actual skipping or including happens at the rule's check, so I'd just name the variable `var_remote_paths`, (or filesystems), and describe it as an aggregator of paths in the system known to be pointing to remote file system.
If there would be rules dedicated to checking just remote filesystems the same variable can be used.

ggbecker replied:
I also would prefer to have this variable applied only to a set of rules where it really makes sense to apply. If one wants to skip a path for every rule that would make sense to be implemented on the scanner side, as IIRC OpenSCAP already does that through environment variables. But that still leaves us with the problem described in another comment: https://github.com/ComplianceAsCode/content/pull/5845#issuecomment-652530450

matejak replied:
Although I am not aware of a possibility to exclude a directory tree by "talking" directly to the scanner. Conversely, we are unable to guarantee that the whole set of applicable rules will get this functionality, so they can be influenced by this variable.
On the other hand, if you have a problem that a network directory is scanned, and you would like to prevent it, having to configure many variables in the same way seems to be user-unfriendly.

I see two options:
1. Multiple (probably < 10) variables for rules that scan the whole local filesystem by default.
2. One variable that is used by rules that are affected by the previous point as well.

ggbecker commented:
> 
> The PR doesn't work out of the box - please help: for some reason, the build system doesn't add the check-export element to the rule XCCDF, without which the external variable used in the OVAL doesn't work. If I add the `check-export` manually into the datastream, it starts to work.

I have done an extensive debugging on why the `check-export` is not appearing in the final datastream and I have concluded the following:

The function https://github.com/ComplianceAsCode/content/blob/master/ssg/parse_oval.py#L46 is not able to parse properly the XML element `<filter>` (https://github.com/ComplianceAsCode/content/pull/5845/files#diff-aab0c1528b29c82f12fd468fa42803daR28). This way the recursion doesn't go all the way down into the `<external_value>` element so it can't find the variable definition.

To solve this we would have to extend the function to recognize that `<filter>` tag can also define a `state` in its text field, then look for the correct _child_ using the text to be able to find the `external_variable` definition.

mildas commented:
**Changes identified:**
Rule file_permissions_unauthorized_world_writable:
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
 Node deleted from OVAL check.
 Node moved within OVAL check.
 Attribute value changed in OVAL check.
 Deleted attribute from OVAL check.
 New node inserted to OVAL check.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_world_writable

yuumasato commented:
> What I don't know is whether this filtering could prevent scanning (which would be really nice), or just collecting, i.e. the scanner would scan, but throw away findings and not eat more RAM, or neither, when the scanner scans, collects and waits for the end to throw everything away.

One way to check this would be by looking at the output of OVAL collected items.

mildas commented:
**Changes identified:**
Rule file_permissions_unauthorized_world_writable:
 Text changed in OVAL check.
 Attribute value changed in OVAL check.
 Deleted attribute from OVAL check.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_world_writable

matejak commented:
Thanks @ggbecker for your investigation, the fix was really easy to implement once the issue has been identified.

mildas commented:
**Changes identified:**
Rule file_permissions_unauthorized_sgid:
 Text changed in OVAL check.
 Text added outsite tags in OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_sgid/oval/shared.xml.
 New node inserted to OVAL check.
Rule file_permissions_unauthorized_world_writable:
 Node moved within OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/oval/shared.xml.
 Text added outsite tags in OVAL check.
 Deleted attribute from OVAL check.
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Attribute value changed in OVAL check.
 Node deleted from OVAL check.
Macro add_nonlocal_state:
 In OVAL check for file_permissions_unauthorized_sgid.
 In OVAL check for file_permissions_unauthorized_world_writable.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_sgid
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_world_writable

matejak commented:
I have changed the variable concept, and introduced it to one more rule. Do you like the direction in which this is heading?
There is a couple of more rules that use the concept of local filesystems, but that would be just some copy-pasting.

matejak commented:
/test e2e-aws-rhcos4-e8
/test e2e-aws-rhcos4-moderate

jan-cerny commented on linux_os/guide/system/var_nonlocal_paths.var:
```diff
@@ -0,0 +1,16 @@
+title: 'Explicit non-local paths'
+
+description: |-
+    Colon-separated list of paths that are not local and shouldn't be scanned if the OVAL check specifies <code>recurse_file_system="local"</code>
```
I wouldn't mention `recurse_file_system=local`. That's a different mechanism which is performed at different level. What your OVAL does is filtering based on states. I would describe it this way:
Colon-separated list of paths that are not local and that will be filtered out and will not be scanned.

jan-cerny commented on linux_os/guide/system/var_nonlocal_paths.var:
```diff
@@ -0,0 +1,16 @@
+interactive: false
+
+options:
+    default: "/sys:/tmp:/proc"
```
The description talks about removing non-local paths, but the default values are local file systems (however special). Maybe try a different wording in the description?

yuumasato commented on linux_os/guide/system/var_nonlocal_paths.var:
```diff
@@ -0,0 +1,16 @@
+documentation_complete: true
+
+title: 'Explicit non-local paths'
```
In general, I find `non-local paths` confusing.
`remote paths` seems clearer to me.

jan-cerny replied:
@matejak What do you think about this? ^

mildas commented:
**Changes identified:**
Rule file_permissions_unauthorized_sgid:
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_sgid/oval/shared.xml.
 Deleted attribute from OVAL check.
 New node inserted to OVAL check.
 Text added outsite tags in OVAL check.
 Node moved within OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
 Node deleted from OVAL check.
Rule file_permissions_unauthorized_world_writable:
 Deleted attribute from OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/oval/shared.xml.
 New node inserted to OVAL check.
 Text added outsite tags in OVAL check.
 Node moved within OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
 Node deleted from OVAL check.
Macro add_remote_state:
 In OVAL check for file_permissions_unauthorized_world_writable.
 In OVAL check for file_permissions_unauthorized_sgid.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_sgid
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_world_writable

jan-cerny commented:
/test e2e-aws-rhcos4-moderate

matejak commented:
/retest

yuumasato reviewed (commented):
LGTM, I just a have a concern about ignoring paths by default.

redhatrises commented on linux_os/guide/system/permissions/files/dir_perms_world_writable_sticky_bits/rule.yml:
```diff
@@ -18,6 +18,9 @@ description: |-
     following command:
     <pre>$ sudo chmod +t <i>DIR</i></pre>
 
+    You can use the <code>var_remote_paths</code> XCCDF value to specify a colon-separated list of directories that you are sure that are remote.
+    For this profile, that would be: {{{ sub_var_value("var_remote_paths") }}}
```
```suggestion
    The <code>var_remote_paths</code> XCCDF value can be used to specify a colon-separated list of directories that are remote filesystem paths.
    For this profile, <code>var_remote_paths</code> is set to: {{{ sub_var_value("var_remote_paths") }}}
```

matejak replied:
Thank you for those suggestions, I really appreciate that!

redhatrises commented on linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned/rule.yml:
```diff
@@ -12,6 +12,9 @@ description: |-
     Following this, the files should be deleted or assigned to an
     appropriate group.
 
+    You can use the <code>var_remote_paths</code> XCCDF value to specify a colon-separated list of directories that you are sure that are remote.
+    For this profile, that would be: {{{ sub_var_value("var_remote_paths") }}}
```
```suggestion
The <code>var_remote_paths</code> XCCDF value can be used to specify a colon-separated list of directories that are remote filesystem paths.
    For this profile, <code>var_remote_paths</code> is set to: {{{ sub_var_value("var_remote_paths") }}}
```

redhatrises commented on linux_os/guide/system/permissions/files/file_permissions_unauthorized_sgid/rule.yml:
```diff
@@ -16,6 +16,9 @@ description: |-
     the software should be considered an approved package on the system.
     Any SGID file not deployed through an RPM will be flagged for further review.
 
+    You can use the <code>var_remote_paths</code> XCCDF value to specify a colon-separated list of directories that you are sure that are remote.
+    For this profile, that would be: {{{ sub_var_value("var_remote_paths") }}}
```
```suggestion
    The <code>var_remote_paths</code> XCCDF value can be used to specify a colon-separated list of directories that are remote filesystem paths.
    For this profile, <code>var_remote_paths</code> is set to: {{{ sub_var_value("var_remote_paths") }}}
```

redhatrises commented on linux_os/guide/system/permissions/files/file_permissions_unauthorized_suid/rule.yml:
```diff
@@ -16,6 +16,9 @@ description: |-
     the software should be considered an approved package on the system.
     Any SUID file not deployed through an RPM will be flagged for further review.
 
+    You can use the <code>var_remote_paths</code> XCCDF value to specify a colon-separated list of directories that you are sure that are remote.
+    For this profile, that would be: {{{ sub_var_value("var_remote_paths") }}}
```
```suggestion
    The <code>var_remote_paths</code> XCCDF value can be used to specify a colon-separated list of directories that are remote filesystem paths.
    For this profile, <code>var_remote_paths</code> is set to: {{{ sub_var_value("var_remote_paths") }}}
```

redhatrises commented on linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/rule.yml:
```diff
@@ -11,6 +11,9 @@ description: |-
     this applies to real files and not virtual files that are a part of
     pseudo file systems such as <tt>sysfs</tt> or <tt>procfs</tt>.
 
+    You can use the <code>var_remote_paths</code> XCCDF value to specify a colon-separated list of directories that you are sure that are remote.
+    For this profile, that would be: {{{ sub_var_value("var_remote_paths") }}}
```
```suggestion
    The <code>var_remote_paths</code> XCCDF value can be used to specify a colon-separated list of directories that are remote filesystem paths.
    For this profile, <code>var_remote_paths</code> is set to: {{{ sub_var_value("var_remote_paths") }}}
```

redhatrises commented on linux_os/guide/system/permissions/files/no_files_unowned_by_user/rule.yml:
```diff
@@ -10,6 +10,9 @@ description: |-
     Following this, the files should be deleted or assigned to an
     appropriate user.
 
+    You can use the <code>var_remote_paths</code> XCCDF value to specify a colon-separated list of directories that you are sure that are remote.
+    For this profile, that would be: {{{ sub_var_value("var_remote_paths") }}}
```
```suggestion
    The <code>var_remote_paths</code> XCCDF value can be used to specify a colon-separated list of directories that are remote filesystem paths.
    For this profile, <code>var_remote_paths</code> is set to: {{{ sub_var_value("var_remote_paths") }}}
```

redhatrises commented on linux_os/guide/system/var_remote_paths.var:
```diff
@@ -0,0 +1,16 @@
+documentation_complete: true
+
+title: 'Explicit remote paths'
```
```suggestion
title: 'Explicit remote filesystem paths'
```

matejak commented on linux_os/guide/system/var_remote_paths.var:
```diff
@@ -0,0 +1,16 @@
+interactive: false
+
+options:
+    default: "/mnt:/net"
```
I understand your concerns - while it is a "mostly good default", it may not be appropriate for everybody, which is a deal breaker. 
Note: We will need to enhance the test suite in order to be able to test this rule.

mildas commented:
**Changes identified:**
Rule dir_perms_world_writable_sticky_bits:
 Node moved within OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/dir_perms_world_writable_sticky_bits/oval/shared.xml.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
Rule dir_perms_world_writable_system_owned:
 Node moved within OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned/oval/shared.xml.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
Rule file_permissions_unauthorized_sgid:
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_sgid/oval/shared.xml.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
 Attribute value changed in OVAL check.
Rule file_permissions_unauthorized_suid:
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_suid/oval/shared.xml.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
 Attribute value changed in OVAL check.
Rule file_permissions_unauthorized_world_writable:
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/oval/shared.xml.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
 Attribute value changed in OVAL check.
Rule no_files_unowned_by_user:
 Templatization usage changed in /linux_os/guide/system/permissions/files/no_files_unowned_by_user/oval/shared.xml.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
 Attribute value changed in OVAL check.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_start_tls.
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
Macro add_remote_state:
 In OVAL check for dir_perms_world_writable_sticky_bits.
 In OVAL check for file_permissions_unauthorized_sgid.
 In OVAL check for no_files_unowned_by_user.
 In OVAL check for dir_perms_world_writable_system_owned.
 In OVAL check for file_permissions_unauthorized_suid.
 In OVAL check for file_permissions_unauthorized_world_writable.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml dir_perms_world_writable_sticky_bits
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml dir_perms_world_writable_system_owned
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_sgid
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_suid
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_world_writable
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml no_files_unowned_by_user

yuumasato commented:
@openscap-ci test this please

mildas commented:
**Changes identified:**
Rule dir_perms_world_writable_sticky_bits:
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Deleted attribute from OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/dir_perms_world_writable_sticky_bits/oval/shared.xml.
 Node moved within OVAL check.
 Text changed in OVAL check.
Rule dir_perms_world_writable_system_owned:
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned/oval/shared.xml.
 Deleted attribute from OVAL check.
 Node moved within OVAL check.
 Text changed in OVAL check.
Rule file_permissions_unauthorized_sgid:
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_sgid/oval/shared.xml.
 Deleted attribute from OVAL check.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
Rule file_permissions_unauthorized_suid:
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Deleted attribute from OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_suid/oval/shared.xml.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
Rule file_permissions_unauthorized_world_writable:
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Deleted attribute from OVAL check.
 Node moved within OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/oval/shared.xml.
 New node inserted to OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
Rule no_files_unowned_by_user:
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Deleted attribute from OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/no_files_unowned_by_user/oval/shared.xml.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_start_tls.
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
Macro add_remote_state:
 In OVAL check for dir_perms_world_writable_sticky_bits.
 In OVAL check for file_permissions_unauthorized_sgid.
 In OVAL check for file_permissions_unauthorized_suid.
 In OVAL check for no_files_unowned_by_user.
 In OVAL check for file_permissions_unauthorized_world_writable.
 In OVAL check for dir_perms_world_writable_system_owned.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml dir_perms_world_writable_sticky_bits
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml dir_perms_world_writable_system_owned
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_sgid
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_suid
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_world_writable
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml no_files_unowned_by_user

mildas commented:
**Changes identified:**
Rule dir_perms_world_writable_sticky_bits:
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Text added outsite tags in OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/dir_perms_world_writable_sticky_bits/oval/shared.xml.
Rule dir_perms_world_writable_system_owned:
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned/oval/shared.xml.
 Text added outsite tags in OVAL check.
Rule file_permissions_unauthorized_sgid:
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_sgid/oval/shared.xml.
 Text added outsite tags in OVAL check.
Rule file_permissions_unauthorized_suid:
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_suid/oval/shared.xml.
 Text added outsite tags in OVAL check.
Rule file_permissions_unauthorized_world_writable:
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/oval/shared.xml.
 Text added outsite tags in OVAL check.
Rule no_files_unowned_by_user:
 Text changed in OVAL check.
 Templatization usage changed in /linux_os/guide/system/permissions/files/no_files_unowned_by_user/oval/shared.xml.
 New node inserted to OVAL check.
 Text added outsite tags in OVAL check.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
 In Bash remediation for sssd_ldap_start_tls.
Macro add_remote_state:
 In OVAL check for dir_perms_world_writable_sticky_bits.
 In OVAL check for file_permissions_unauthorized_world_writable.
 In OVAL check for no_files_unowned_by_user.
 In OVAL check for file_permissions_unauthorized_suid.
 In OVAL check for dir_perms_world_writable_system_owned.
 In OVAL check for file_permissions_unauthorized_sgid.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml dir_perms_world_writable_sticky_bits
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml dir_perms_world_writable_system_owned
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_sgid
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_suid
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml file_permissions_unauthorized_world_writable
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml no_files_unowned_by_user

jan-cerny commented:
/retest

jan-cerny commented:
/retest

jan-cerny commented on linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/oval/shared.xml:
```diff
@@ -35,4 +36,6 @@
     <unix:filepath operation="pattern match">^/sys/.*$</unix:filepath>
   </unix:file_state>
+
+  {{{ add_remote_state("state_file_in_remote_dir_" + rule_id, "linux", "rpmverifyfile_state") }}}
```
I'd expect a `file_state` here. Am I missing something or is it a mistake?

jan-cerny reviewed (changes requested):
Please fix the failing test:

```
[jcerny@thinkpad scap-security-guide{pr/5845}]$ python3 tests/test_suite.py rule  --libvirt qemu:///system ssgts_rhel8 --datastream /usr/share/xml/scap/ssg/content/ssg-rhel8-ds.xml file_permissions_unauthorized_world_writable
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2020-10-05-1643/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_file_permissions_unauthorized_world_writable
INFO - Script wwritable_in_root.fail.sh using profile (all) OK
WARNING - No remediation is available for rule 'xccdf_org.ssgproject.content_rule_file_permissions_unauthorized_world_writable'.
ERROR - Script wwritable_in_mnt.pass.sh using profile (all) found issue:
ERROR - Rule evaluation resulted in fail, instead of expected pass during initial stage 
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_file_permissions_unauthorized_world_writable'.
[jcerny@thinkpad scap-security-guide{pr/5845}]$ 
```

matejak commented:
Fixing the test requires enhancement of the test suite, as no directory is considered remote by default.

jan-cerny commented:
We probably shouldn't merge a failing test because the rule  file_permissions_unauthorized_world_writable is part of multiple profiles.

matejak commented:
#6764 raises a good point in the description - the solution of excluding paths using states is inefficient, because objects have to be collected anyway, and then thrown out.
